### ISSUE TITLE
allow setting envoy proxy resource when using "expose" stanza

### ISSUE THREAD

**Issue by fredwangwang** [hashicorp/nomad#9854]
### Nomad version
v1.0.1

### Operating system and Environment details
Linux

### Issue

The default resource usage for envoy_sidecar is a bit too high for us, so we relying on `sidecar_task` to lower the resource:
```hcl
sidecar_task {
    resources {
      cpu    = ...
      memory = ...
    }
  }
```
However, we found out once this is set, we are not able to set [`expose =true`](https://www.nomadproject.io/docs/job-specification/expose) anymore:
![image](https://user-images.githubusercontent.com/21690857/105099436-ed580780-5a68-11eb-8f63-0e13672f62a6.png)

So for now we have to remove the resource override for the envoy sidecar. But it really should be able to have expose and resource setting for sidecar at the same time.

Related line here:
https://github.com/hashicorp/nomad/blob/9a36ebe2289ceed706bc5dc97472c4e697a9df13/nomad/job_endpoint_hook_expose_check.go#L158-L179

**Comment by idrennanvmware** [comment]
Just touching base here.  We are getting crushed in deployment resource requirements with mesh enabled and this feature on.   Just deployment of a single service with 4 instances is taking 1000mhz of cpu

This is one of those things that get exponentially worse with every service that goes to the mesh

**Comment by evandam** [comment]
Agreed this seems really important to be able to set. Alternatively, is there any way that the default resources can be set in the Nomad/Consul config so each job wouldn't have to set this?

**Comment by idrennanvmware** [comment]
@evandam  

In our case we have different services with different levels of requests per second, we would want each service having it's own settings (and we encourage our teams to justify their resource usage) - but I can see the usefulness of a way to override the defaults at the global level.

**Comment by idrennanvmware** [comment]
I need to take a look deeper in to the code, but a thought struck me.   Can it not be checked if the sidecar task is using the container specified in ```meta.connect.sidecar_image``` as part of the validation?  If that value matches the image then overrides are "safe" at the resource level

**Comment by shoenig** [comment]
I think we can just remove the precondition altogether; this was originally added when we still had hope of Connect potentially being backed by more than just Envoy, but that seems unlikely now. `expose` is a leaky abstraction of an Envoy feature, which is why we were trying to be careful here.

I'll work on this today so we can slip the fix into 1.0.4

**Comment by idrennanvmware** [comment]
Thank you!   This is a huge improvement for us!
